[quant][graph] Update quantize_dynamic_script API to take sample model args

supriyar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #36844 [quant][graph] Add quantized::mul_relu and quantized::mul_scalar_relu ops
* #36818 [quant][graph] Support for quantized::mul and quantized::mul_scalar
* **#36817 [quant][graph] Update quantize_dynamic_script API to take sample model args**

Summary:
For dynamic quant we need to run the observers on the weights to calculate the qparams before calling convert on the model.
The API requires the user to provide dummy inputs that will be fed to the model after the prepare step to run the observers

Test Plan:
test_quanitze_script.py
test_quantization.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21134439](https://our.internmc.facebook.com/intern/diff/D21134439)

supriyar committed:
[quant][graph] Update quantize_dynamic_script API to take sample model args
Summary:
For dynamic quant we need to run the observers on the weights to calculate the qparams before calling convert on the model.
The API requires the user to provide dummy inputs that will be fed to the model after the prepare step to run the observers

Test Plan:
test_quanitze_script.py
test_quantization.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar committed:
Update on "[quant][graph] Update quantize_dynamic_script API to take sample model args"


Summary:
For dynamic quant we need to run the observers on the weights to calculate the qparams before calling convert on the model.
The API requires the user to provide dummy inputs that will be fed to the model after the prepare step to run the observers

Test Plan:
test_quanitze_script.py
test_quantization.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented on torch/quantization/_quantize_script.py:
```diff
@@ -133,5 +133,5 @@ def quantize_script(model, qconfig_dict, run_fn, run_args, inplace=False, debug=
-def quantize_dynamic_script(model, qconfig_dict, run_fn, run_args, debug=False):
-    return _quantize_script(model, qconfig_dict, run_fn, run_args, is_dynamic=True, debug=debug)
+def quantize_dynamic_script(model, qconfig_dict, sample_model_inputs, debug=False):
+    return _quantize_script(model, qconfig_dict, None, sample_model_inputs, is_dynamic=True, debug=debug)
```
can we provide some default arguments for `run_fn`, `run_args`, `is_dynamic`, `debug` in `_quantize_script` and pass `sample_model_inputs, is_dynamic=True, debug=debug` by keyword so we don't need to pass in `None` here?

supriyar replied:
sure

supriyar committed:
Update on "[quant][graph] Update quantize_dynamic_script API to take sample model args"


Summary:
For dynamic quant we need to run the observers on the weights to calculate the qparams before calling convert on the model.
The API requires the user to provide dummy inputs that will be fed to the model after the prepare step to run the observers

Test Plan:
test_quanitze_script.py
test_quantization.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]
